### PR TITLE
Webpack quality of life improvements for dev mode

### PR DESCRIPTION
**Pull request by jakolehm**
- removes `webpack-dev-server` dependency from the application code (it was runtime import in `src/main/index.ts`), webpack-dev-server is now running as a separate process
- proxies asset requests to webpack-dev-server in development mode
- adds filesystem cache for faster boot time
- removes unnecessary `devDependencies` from bundled extensions
- simplifies Makefile

### CODE REVIEW

File: src/main/router.ts [hunk 1]
```diff
@@ -140,12 +143,26 @@ export class Router {
...
       }
     }
-
   }
 
   protected addRoutes() {
     // Static assets
-    this.router.add({ method: "get", path: "/{path*}" }, Router.handleStaticFile);
+    if (this.dependencies.httpProxy) {
+      this.router.add({ method: "get", path: "/{path*}" }, (apiReq: LensApiRequest) => {
```
Comment (jakolehm) [inline src/main/router.ts:151]: This is ugly but I didn't want to polish further because it will be refactored anyway in #4859 .

**Review by nevalla — DISMISSED** [review]

**Review by nevalla — APPROVED** [review]

File: extensions/kube-object-event-status/package.json [hunk 2]
```diff
@@ -8,18 +8,15 @@
...
   "files": [
     "dist/**/*"
   ],
-  "dependencies": {},
   "devDependencies": {
     "@k8slens/extensions": "file:../../src/extensions/npm/extensions",
-    "ts-loader": "latest",
-    "typescript": "latest",
-    "webpack": "latest"
+    "npm": "^8.5.3"
```
Comment (jakolehm) [inline extensions/kube-object-event-status/package.json:20]: npx in npm6 is buggy, that's why bundled extensions now have npm@8

Comment (Nokel81) [reply extensions/kube-object-event-status/package.json:20]: Do we need `npx`?

File: package.json [hunk 3]
```diff
@@ -15,10 +15,9 @@
     "dev": "concurrently -i -k \"yarn run dev-run -C\" yarn:dev:*",
     "dev-build": "concurrently yarn:compile:*",
     "debug-build": "concurrently yarn:compile:main yarn:compile:extension-types",
-    "dev-run": "nodemon --watch static/build/main.js --exec \"electron --remote-debugging-port=9223 --inspect .\"",
+    "dev-run": "nodemon --watch ./static/build/main.js --exec \"electron --remote-debugging-port=9223 --inspect .\"",
     "dev:main": "yarn run compile:main --watch --progress",
-    "dev:renderer": "yarn run compile:renderer --watch --progress",
-    "dev:extension-types": "yarn run compile:extension-types --watch --progress",
```
Comment (Nokel81) [inline package.json:21]: Why do we not run this anymore?

File: extensions/kube-object-event-status/package.json (hunk 2 above)
Comment (jakolehm) [reply extensions/kube-object-event-status/package.json:20]: `npx` is the thing that allows us to remove webpack/typescript/ts-loader etc.

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**